provider/aws/aws.go

```diff
@@ -333,22 +355,41 @@ func (p *AWSProvider) Zones(ctx context.Context) (map[string]*route53.HostedZone
 				}
 			}
 
-			zones[aws.StringValue(zone.Id)] = zone
+			zones[aws.StringValue(zone.Id)] = &profiledZone{
+				profile: profile,
+				zone:    zone,
+			}
 		}
 
 		return true
 	}
 
-	err := p.client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
-	if err != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list hosted zones: %w", err))
-	}
-	if tagErr != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list zones tags: %w", tagErr))
+	for p, client := range p.clients {
+		profile = p
+		err := client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
+		if err != nil {
+			var awsErr awserr.Error
+			if errors.As(err, &awsErr) {
+				switch awsErr.Code() {
+				case "AccessDenied":
+					log.Warnf("Skipping AWS profile %q due to missing permission: %v", profile, awsErr.Message())
+					continue
+				case "InvalidClientTokenId", "ExpiredToken", "SignatureDoesNotMatch":
+					log.Warnf("Skipping AWS profile %q due to credential issues: %v", profile, awsErr.Message())
```

Comment:
Why this is only a warning and not fatal?
This does not seem to be a temporary issue, but requires attention so we should stop the process.

At least I expect this to be an error instead of warning.

Reply:
Hi @szuecs, thanks for the review. 

In our use-case where we handle many AWS profiles with one external-dns instance, it occasionally happens that one profile is not ready due to eventual consistency topics during resource provisioning. However, the other profiles are working well in those situations. That's why we don't want the process to be killed. Especially because we heavily depend on caching in order not to face the Route53 throttling thresholds. Thats why I implemented it as a warning.

However, I understand other views on that. Changing this two cases to log an error instead of a warning would be completely fine for me. Additionally I see two more possibilities to handle that:

1. In order to keep the former semantics I could log the warnings  (or errors) and add fatal behavior in case none of the given profiles is working.
2. I can introduce a behavioral switch as command line argument in order to let the user decide between fatal and graceful behavior (don't like that option much TBH). 

WDYT?

Reply:
Situation:
You have external-dns running and you deploy a new version.
The new version can now only work on half of the profiles because of some bug.
I think what we should do is to fatal and let the operator decide what she wants to do. 

I would expect that they would execute a rollback and try to understand what failed.
Maybe they can also remove some config for profiles that do not work and fix later these or they just roll back. 
If we only log it means we ignore a problem that does not magically heal itself on a retry.

Throttling case can be detected and handled on our side to not fatal.

Reply:
I do not agree 100% but I understand your opinion. I changed it to what you're proposing. Also added the throttling case too and going to take further care on throttling for all the other API calls external-dns is leveraging - but with separate PR.